Sources/AppStoreConnect/Networking/RetryStrategy.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+
+package func retry<T>(with strategy: some RetryStrategy, _ block: @Sendable () async throws -> T) async rethrows -> T {
+    var iteration = 0
+
+    while true {
+        do {
+            return try await block()
+        } catch {
+            guard try await strategy.waitAndContinue(for: error, iteration: iteration) else { throw error }
+        }
+
+        iteration += 1
+    }
+}
+
+/// Execute a waiting algorithm and inform the client whether to continue retrying a request.
+///
+/// This enum is intended to be provided as input to the methods on ``AppStoreConnect``.
+public protocol RetryStrategy: Sendable {
+    /// Respond to an error from ``AppStoreConnectClient``, typically by waiting for a period of time.
+    /// - Parameter error: The received error.
+    /// - Returns: A boolean indicating whether or not to continue retrying.
+    func waitAndContinue(for error: any Error, iteration: Int) async throws -> Bool
+}
+
+/// The strategy that never retries
+public struct NeverRetryStrategy: RetryStrategy {
+    public func waitAndContinue(for error: any Error, iteration: Int) async throws -> Bool { false }
+}
+
+extension RetryStrategy where Self == NeverRetryStrategy {
+    public static var never: Self { Self() }
+}
+
+/// The strategy that retries on a fixed interval, in seconds, until a limit is reached.
+public struct FixedIntervalRetryStrategy<Clock: _Concurrency.Clock, Duration: Swift.DurationProtocol>: RetryStrategy
+where Duration == Clock.Duration {
+    var interval: Duration
+    var limit: Int
+    var clock: Clock
+
+    public init(interval: Duration, limit: Int, clock: Clock) {
+        self.interval = interval
+        self.limit = limit
+        self.clock = clock
+    }
+
+    public func waitAndContinue(for error: any Error, iteration: Int) async throws -> Bool {
+        guard iteration < limit else { return false }
+
+        try await Task.sleep(for: interval, clock: clock)
+
+        return true
+    }
+}
+
+extension RetryStrategy {
+    /// The strategy that retries on a fixed interval, in seconds, until a limit is reached.
+    static func fixedInterval<Clock: _Concurrency.Clock, Duration: Swift.DurationProtocol>(
+        _ interval: Duration,
+        limit: Int,
+        clock: Clock
+    ) -> FixedIntervalRetryStrategy<Clock, Duration>
+    where Self == FixedIntervalRetryStrategy<Clock, Duration> {
+        Self(interval: interval, limit: limit, clock: clock)
+    }
+}
+
+extension RetryStrategy where Self == FixedIntervalRetryStrategy<ContinuousClock, ContinuousClock.Duration> {
```

Comment:
No need for [SuspendingClock](https://developer.apple.com/documentation/swift/suspendingclock) default?, seems not needed 🎉